### PR TITLE
Txqueueasyncverify

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: libethereum/TransactionQueue.cpp [hunk 1]
```diff
@@ -31,12 +31,34 @@ using namespace dev::eth;
 const char* TransactionQueueChannel::name() { return EthCyan "┉┅▶"; }
 const char* TransactionQueueTraceChannel::name() { return EthCyan " ┅▶"; }
 
+TransactionQueue::TransactionQueue()
+{
+	// Allow some room for other activity
+	unsigned verifierThreads = std::max(thread::hardware_concurrency(), 3U) - 2U;
```
Comment (arkpar) [inline libethereum/TransactionQueue.cpp]: do we really need more than one thread here?

File: libethereum/TransactionQueue.cpp [hunk 2]
```diff
@@ -252,16 +336,20 @@ void TransactionQueue::noteGood(std::pair<h256, Transaction> const& _t)
...
-void TransactionQueue::drop(h256 const& _txHash)
+void TransactionQueue::drop(h256 const& _txHash, ImportResult _ir)
 {
 	UpgradableGuard l(m_lock);
 
 	if (!m_known.count(_txHash))
 		return;
 
+	if (m_callbacks.count(_txHash) && m_callbacks[_txHash])
+		m_callbacks[_txHash](_ir);
```
Comment (arkpar) [inline libethereum/TransactionQueue.cpp]: These callbacks don't seem to be used currently

Comment (gavofyork) [inline libethereum/TransactionQueue.cpp]: indeed - they probably will be in the future though.

File: libethereum/TransactionQueue.cpp (hunk 1 above)
Comment (gavofyork) [inline libethereum/TransactionQueue.cpp]: don't really see any harm. if there's 256 incoming transactions, they'll help and if there's only one, then 5 will never get woken up anyway.

**Pull request merged**